server/clip_server/executors/clip_torch.py

```diff
@@ -108,57 +119,92 @@ async def rank(self, docs: 'DocumentArray', parameters: Dict, **kwargs):
-                    ),
-                    batch_size=self._minibatch_size,
-                    pool=self._pool,
-                ):
-                    with self.monitor(
-                        name='encode_images_seconds',
-                        documentation='images encode time in seconds',
-                    ):
-                        minibatch.embeddings = (
-                            self._model.encode_image(**batch_data)
-                            .cpu()
-                            .numpy()
-                            .astype(np.float32)
-                        )
-
-            # for text
-            if _txt_da:
-                for minibatch, batch_data in _txt_da.map_batch(
-                    self._preproc_texts,
-                    batch_size=self._minibatch_size,
-                    pool=self._pool,
-                ):
-                    with self.monitor(
-                        name='encode_texts_seconds',
-                        documentation='texts encode time in seconds',
-                    ):
-                        minibatch.embeddings = (
-                            self._model.encode_text(**batch_data)
-                            .cpu()
-                            .numpy()
-                            .astype(np.float32)
-                        )
+    async def encode(
+        self,
+        docs: 'DocumentArray',
+        tracing_context=None,
+        parameters: Dict = {},
+        **kwargs,
+    ):
+        with self.tracer.start_as_current_span(
+            'encode', context=tracing_context
+        ) as span:
+            span.set_attribute('device', self._device)
+            span.set_attribute('runtime', 'torch')
+            access_paths = parameters.get('access_paths', self._access_paths)
+            if 'traversal_paths' in parameters:
+                warnings.warn(
+                    f'`traversal_paths` is deprecated. Use `access_paths` instead.'
+                )
+                access_paths = parameters['traversal_paths']
+            _drop_image_content = parameters.get('drop_image_content', False)
+
+            _img_da = DocumentArray()
+            _txt_da = DocumentArray()
+            for d in docs[access_paths]:
+                split_img_txt_da(d, _img_da, _txt_da)
+
+            with self.tracer.start_as_current_span('inference') as inference_span:
+                with torch.inference_mode():
+                    inference_span.set_attribute(
+                        'drop_image_content', _drop_image_content
+                    )
+                    inference_span.set_attribute('minibatch_size', self._minibatch_size)
+                    inference_span.set_attribute(
+                        'has_img_da', True if _img_da else False
+                    )
+                    inference_span.set_attribute(
+                        'has_txt_da', True if _txt_da else False
+                    )
```

Comment:
Do we need these in onnx & tensorrt as well?

Reply:
I will update the code.